docusaurus/docs/iOS/guides/working-with-attachments.md

```diff
@@ -264,3 +252,57 @@ controller.createNewMessage(text: "work-out-test", attachments: [.init(payload:
     print("test message was added")
 }
 ```
+
+In case you need to interact with your custom attachment, there are a couple of steps required:
+1. Create a delegate for your custom attachment view which extends from `ChatMessageContentViewDelegate`.
+2. Create a custom `ChatMessageListVC` if you didn't already, and make it conform to the delegate created in step 1.
+3. Change your custom injector and add a tap gesture recognizer to your custom view. The delegate can be called by accessing `contentView.delegate` and casting it to your custom delegate.
+
+Below is the full example on how to add a interaction to the custom workout attachment:
+
+```swift
+// Step 1
+protocol WorkoutAttachmentViewDelegate: ChatMessageContentViewDelegate {
+    func didTapOnWorkoutAttachment(
+        _ attachment: ChatMessageWorkoutAttachment
+    )
+}
+
+// Step 2
+class CustomChatMessageListVC: ChatMessageListVC, WorkoutAttachmentViewDelegate {
+    func didTapOnWorkoutAttachment(_ attachment: ChatMessageWorkoutAttachment) {
+        // For example, here you can present a view controller to display the workout
+        let workoutViewController = WorkoutViewController(workout: attachment)
+        navigationController?.pushViewController(workoutViewController, animated: true)
+    }
+}
+
+// Step 3
+class WorkoutAttachmentViewInjector: AttachmentViewInjector {
+    let workoutView = WorkoutAttachmentView()
+
+    override open func contentViewDidLayout(options: ChatMessageLayoutOptions) {
```

Comment:
> Let's ask integrators to call super unless there's a strong reason not to do that

It's the rule of thumb. We should make devs used to always call `super` as the first thing when customizing components, I think. 

In this case, super does nothing but in other places, it can break things. And yeah, super might spot being empty in the future

Reply:
Done 👍